File: tests/test_cdc/update_replicate_config/quick_start.go
package main

import (
	"context"
	"fmt"
	"log"
	"time"

	"github.com/milvus-io/milvus/client/v2/milvusclient"
)

func main() {
	const (
		clusterAAddr = `http://127.0.0.1:19530`
		clusterBAddr = `http://127.0.0.1:19531`
		clusterAID   = `cluster-a`
		clusterBID   = `cluster-b`
		pchannelNum  = 16
	)

	clsuterAPchannels := make([]string, 0, pchannelNum)
	clusterBPchannels := make([]string, 0, pchannelNum)
	for i := 0; i < pchannelNum; i++ {
		clsuterAPchannels = append(clsuterAPchannels, fmt.Sprintf("%s-rootcoord-dml_%d", clusterAID, i))
		clusterBPchannels = append(clusterBPchannels, fmt.Sprintf("%s-rootcoord-dml_%d", clusterBID, i))
	}
	clusterA := milvusclient.NewMilvusClusterBuilder(clusterAID).
		WithURI(clusterAAddr).
		WithPchannels(clsuterAPchannels...).
		Build()
	clusterB := milvusclient.NewMilvusClusterBuilder(clusterBID).
		WithURI(clusterBAddr).
		WithPchannels(clusterBPchannels...).
		Build()

	config := milvusclient.NewReplicateConfigurationBuilder().
		WithCluster(clusterA).
		WithCluster(clusterB).
		WithTopology(clusterAID, clusterBID).
		Build()

	updateReplicateConfigToCluster := func(clusterAddr string) {
		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
		defer cancel()
		cli, err := milvusclient.New(ctx, &milvusclient.ClientConfig{
			Address: clusterAddr,
		})
		if err != nil {
			log.Fatal(err)
		}
		err = cli.UpdateReplicateConfiguration(ctx, config)
		if err != nil {
			log.Fatal(err)
		}
		log.Printf("Replicate configuration updated to cluster %s.\n", clusterAddr)
	}
	updateReplicateConfigToCluster(clusterAAddr)
	updateReplicateConfigToCluster(clusterBAddr)
}

File: tests/test_cdc/testcases/quick_start.py
from pymilvus import MilvusClient

cluster_a_addr = "http://127.0.0.1:19530"
cluster_b_addr = "http://127.0.0.1:19531"
cluster_a_token = "root:Milvus"
cluster_b_token = "root:Milvus"
cluster_a_id = "cluster-a"
cluster_b_id = "cluster-b"
pchannel_num = 16

cluster_a_pchannels = []
cluster_b_pchannels = []
for i in range(pchannel_num):
    cluster_a_pchannels.append(f"{cluster_a_id}-rootcoord-dml_{i}")
    cluster_b_pchannels.append(f"{cluster_b_id}-rootcoord-dml_{i}")

config = {
    "clusters": [
        {
            "cluster_id": cluster_a_id,
            "connection_param": {
                "uri": cluster_a_addr,
                "token": cluster_a_token
            },
            "pchannels": cluster_a_pchannels
        },
        {
            "cluster_id": cluster_b_id,
            "connection_param": {
                "uri": cluster_b_addr,
                "token": cluster_b_token
            },
            "pchannels": cluster_b_pchannels
        }
    ],
    "cross_cluster_topology": [
        {
            "source_cluster_id": cluster_a_id,
            "target_cluster_id": cluster_b_id
        }
    ]
}

client_a = MilvusClient(uri=cluster_a_addr, token=cluster_a_token)
client_a.update_replicate_configuration(**config)
client_a.close()

client_b = MilvusClient(uri=cluster_b_addr, token=cluster_b_token)
client_b.update_replicate_configuration(**config)
client_b.close()
